server/service/apple_mdm.go

```diff
@@ -1117,15 +1120,26 @@ func (svc *Service) EnqueueMDMAppleCommand(
 					FailedUUIDs: apnsErr.FailedUUIDs,
 				}, nil
 			}
+			// command was not processed on all hosts so return 422
+			err := fleet.NewInvalidArgumentError(
+				"command",
+				fmt.Sprintf("command failed to process on all hosts: %v", err),
+			).WithStatus(http.StatusUnprocessableEntity)
+			return http.StatusUnprocessableEntity, nil, ctxerr.Wrap(ctx, err, "enqueue command")
```

Comment:
do you mind clarifying why this is a client error? for example: let's say the fleet server failed to send the APNs notification to all hosts.

Is this to cover the case of the client sending an invalid host uuid?

Reply:
I think I was misinterpreting `APNSDeliveryError` to mean that the request was rejected by Apple because the command itself was not able to be processed based on the semantic content of the command or that a device uuid isn't valid. What status code seems most appropriate to you?